hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/source/TestIncrementalInputSplits.java

```diff
@@ -191,6 +196,85 @@ void testInputSplitsWithPartitionPruner(
+    HoodieTableFileSystemView fileSystemView =
+            new HoodieTableFileSystemView(metaClient, commitsTimeline, fileStatuses);
+    Map<String, String> file2Instant = fileSystemView.getAllFileSlices("par1")
+            .collect(Collectors.toMap(FileSlice::getFileId, FileSlice::getBaseInstantTime));
+    IncrementalInputSplits iis = IncrementalInputSplits.builder()
+            .conf(conf)
+            .path(new Path(basePath))
+            .rowType(TestConfigurations.ROW_TYPE)
+            .partitionPruner(null)
+            .build();
+
+    IncrementalInputSplits.Result result = iis.inputSplits(metaClient, null, null, false);
+    result.getInputSplits().stream().forEach(split -> {
+      if (file2Instant.containsKey(split.getFileId())) {
+        assertEquals(split.getLatestCommit(), file2Instant.get(split.getFileId()));
+      }
+    });
+  }
+
+  @Test
+  void testUpsertInputSplitsWithSpeedLimit() throws Exception {
+    Configuration conf = TestConfigurations.getDefaultConf(basePath);
+    conf.set(FlinkOptions.READ_AS_STREAMING, true);
+    conf.set(FlinkOptions.READ_STREAMING_SKIP_CLUSTERING, true);
+    conf.set(FlinkOptions.READ_STREAMING_SKIP_COMPACT, true);
+    conf.set(FlinkOptions.OPERATION, WriteOperationType.UPSERT.value());
+
+    // insert data
+    TestData.writeData(TestData.DATA_SET_INSERT, conf);
+    TestData.writeData(TestData.DATA_SET_UPDATE_INSERT, conf);
+    TestData.writeData(TestData.DATA_SET_UPDATE_INSERT, conf);
+    TestData.writeData(TestData.DATA_SET_UPDATE_INSERT, conf);
+
+    HoodieTimeline commitsTimeline = metaClient.reloadActiveTimeline()
+            .filter(hoodieInstant -> hoodieInstant.getAction().equals(HoodieTimeline.DELTA_COMMIT_ACTION));
+    List<HoodieInstant> instants = commitsTimeline.getInstants();
+    List<HoodieCommitMetadata> metadataList = instants.stream()
+            .map(instant -> WriteProfiles.getCommitMetadata(tableName, new Path(basePath), instant, commitsTimeline)).collect(Collectors.toList());
+    FileStatus[] fileStatuses = WriteProfiles.getFilesFromMetadata(new Path(basePath), metaClient.getHadoopConf(), metadataList, metaClient.getTableType());
+    HoodieTableFileSystemView fileSystemView =
+            new HoodieTableFileSystemView(metaClient, commitsTimeline, fileStatuses);
+    Map<String, String> file2Instant = fileSystemView.getAllFileSlices("par1")
+            .collect(Collectors.toMap(FileSlice::getFileId, fileSlice -> fileSlice.getLatestLogFile().isPresent()
+                 ? fileSlice.getLatestLogFile().get().getDeltaCommitTime()
+                 : fileSlice.getBaseInstantTime()));
+    
+    IncrementalInputSplits iis = IncrementalInputSplits.builder()
+            .conf(conf)
+            .path(new Path(basePath))
+            .rowType(TestConfigurations.ROW_TYPE)
+            .partitionPruner(null)
+            .build();
```

Comment:
Did you add a test case where split latest commit not equals the `lastInstant` ?